### PR TITLE
Adapt transaction fees business logic with new transaction structure - Closes #1502 and #1503

### LINKED ISSUE #1502: Adapt `useCalculationFees` hook business logic with new transaction structure
### Description
In order to calculate fees working correctly its need to know the network priority and get raw transaction from transaction class also min fee and max fee logic is must be implemented

### Acceptance Criteria
- Take from reference the current implementation of fees calculation in Mobile.
- Create and adapt the current methods from the `Transaction` class to cover all fees calculations use cases.
- Adopt transaction amount validation based on fee and balance availability
- Refactor and remove old fees calculations code (business logic and integration with UI).
- Integrate this new business logic with existing UI (input and output components).
- Adopt unit test 

### Additional Information
N/A.

### PR DESCRIPTION
**Pull request by ClementeSerrano**
### What was the problem?

This PR resolves #1502 and #1503

### How was it solved?

- [x] Adapt previous *Priority* fields to new transaction structure on send token process.
- [x] Create and adapt the current methods from the `Transaction` class to cover all fees calculations use cases.
- [x] Adopt transaction amount validation based on fee and balance availability
- [x] Refactor and remove old fees calculations code (business logic and integration with UI).
- [x] Integrate this new business logic with existing UI (input and output components).
- [x] Adopt unit test 

### How was it tested?

Locally on iOS & Android Simulators

### CODE REVIEW

File: src/modules/SendToken/components/SummaryStep/hooks.js [hunk 1]
```diff
@@ -35,14 +33,13 @@ export function useSendTokenSummary({ form }) {
...
 
-  const transactionFee = useTransactionFeeCalculator({
-    tokenID,
-    amount,
-    priority,
-    message,
-  });
+  const token = supportedTokensData?.find((_token) => _token.tokenID === tokenID);
+
+  const transactionFee = Lisk.transactions.convertBeddowsToLSK(
```
Comment (soroushm) [inline src/modules/SendToken/components/SummaryStep/hooks.js:40]: its need to be memoized

**Review by soroushm — CHANGES_REQUESTED** [review]

File: src/modules/Transactions/utils/Transaction.js [hunk 2]
```diff
@@ -156,18 +161,25 @@ export class Transaction {
   computeFee() {
```
Comment (ClementeSerrano) [inline src/modules/Transactions/utils/Transaction.js:161]: took https://github.com/LiskHQ/discussions/discussions/66 as ref

File: src/modules/SendToken/components/SummaryStep/hooks.js (hunk 1 above)
Comment (soroushm) [reply src/modules/SendToken/components/SummaryStep/hooks.js:40]: its a computation running by every single re-render while user typing its going to be calculate again

File: src/modules/SendToken/constants/transactions.js [hunk 3]
```diff
@@ -123,31 +107,31 @@ export const transactions = {
   [moduleCommandNameIdMap.transfer]: {
     moduleAssetId: '2:0',
     fee: 1e7,
-    title: t('Transfer'),
+    title: i18next.t('Transfer'),
     image: (theme) => (theme === themes.light ? transferLight : transferDark),
   },
   [moduleCommandNameIdMap.registerMultisignatureGroup]: {
```
Comment (soroushm) [inline src/modules/SendToken/constants/transactions.js:113]: as mobile dosnt have transaction other than transfer do we need to keep other command or schemas on it ?

File: src/modules/SendToken/hooks/useCCMFeeCalculator.js [hunk 4]
```diff
@@ -14,10 +11,7 @@ export default function useCCMFeeCalculator({
 
   useEffect(() => {
     try {
-      const fee = getCCMFee({
-        senderApplicationChainID,
-        recipientApplicationChainID,
-      });
+      const fee = 0;
```
Comment (soroushm) [inline src/modules/SendToken/hooks/useCCMFeeCalculator.js:14]: add ccm fees to the config then we can modify it later

Comment (soroushm) [reply src/modules/SendToken/hooks/useCCMFeeCalculator.js:14]: it dosnt need state 
```
export default function useCCMFeeCalculator({
  senderApplicationChainID,
  recipientApplicationChainID,
}) {
const isCCM = senderApplicationChainID !== recipientApplicationChainID
return  isCCM ? config.CCMFee : 0
}
```

File: src/modules/SendToken/hooks/useInitializationFeeCalculator.js [hunk 5]
```diff
@@ -1,9 +1,8 @@
-/* eslint-disable max-statements */
 import { useEffect, useState } from 'react';
 
 import { useAccounts } from 'modules/Accounts/hooks/useAccounts';
-import { getInitializationFee } from '../helpers';
 
+// TODO: Implement real calculation business logic.
```
Comment (soroushm) [inline src/modules/SendToken/hooks/useInitializationFeeCalculator.js:5]: create a ticket or like ccm fee added to the config you can find it has initialized fee by calling the transaction endpoint with the adress if has transaction the fee is 0 other wise its need to return fee from config

File: src/modules/SendToken/hooks/useTransactionPriorities.js [hunk 6]
```diff
@@ -1,21 +0,0 @@
-import { useMemo } from 'react';
-import { useGetFeesByPriorityQuery } from '../api/useGetFeesByPriorityQuery';
```
Comment (soroushm) [inline src/modules/SendToken/hooks/useTransactionPriorities.js:2]: whats happend to the Priority?

File: src/modules/Transactions/utils/Transaction.js [hunk 7]
```diff
@@ -20,9 +20,12 @@ export class Transaction {
 
   _auth = null;
 
+  _feeEstimatePerByte = null;
+
   transaction = {
     module: null,
     command: null,
+    priority: null,
```
Comment (soroushm) [inline src/modules/Transactions/utils/Transaction.js:28]: priority needs to be update after selection

File: src/modules/SendToken/components/SelectTokenStep/components.js [hunk 8]
```diff
@@ -226,47 +225,18 @@ export function SendTokenMessageField({ value, onChange, style }) {
...
     styles: getSendTokenSelectTokenStepStyles(),
   });
 
-  const shouldShowPrioritiesData = useMemo(
-    () => prioritiesData && prioritiesData[0]?.fee,
-    [prioritiesData]
-  );
+  const priorities =
+    transaction.data._feeEstimatePerByte &&
+    Object.entries(transaction.data._feeEstimatePerByte).map(([code, fee]) => ({ code, fee }));
```
Comment (soroushm) [inline src/modules/SendToken/components/SelectTokenStep/components.js:235]: _feeEstimatePerByte is root element of transaction i dont think its in the data
transaction.data._feeEstimatePerByte

File: src/modules/Transactions/utils/Transaction.js [hunk 9]
```diff
@@ -49,6 +53,7 @@ export class Transaction {
     this.isLoading = false;
     this._networkStatus = networkStatus;
     this._auth = auth;
+    this._feeEstimatePerByte = feeEstimatePerByte;
```
Comment (soroushm) [inline src/modules/Transactions/utils/Transaction.js:56]: `this._` means is private and its not exposed, but some of the utils use it from outside pleas remove the underscore

File: src/modules/SendToken/components/SelectTokenStep/components.js (hunk 8 above)
Comment (soroushm) [reply src/modules/SendToken/components/SelectTokenStep/components.js:235]: its better to use the query hooks directly here other than take it from private resource from transaction object

File: src/modules/SendToken/constants/transactions.js (hunk 3 above)
Comment (ClementeSerrano) [reply src/modules/SendToken/constants/transactions.js:113]: yes, but this hash table is for rendering the transaction information when is already submitted, not for creating them (all of them except token transfer), so we need it on the components

File: src/modules/SendToken/hooks/useCCMFeeCalculator.js (hunk 4 above)
Comment (ClementeSerrano) [reply src/modules/SendToken/hooks/useCCMFeeCalculator.js]: done

File: src/modules/SendToken/components/SelectTokenStep/components.js (hunk 8 above)
Comment (ClementeSerrano) [reply src/modules/SendToken/components/SelectTokenStep/components.js]: done

File: src/modules/Transactions/utils/Transaction.js (hunk 9 above)
Comment (ClementeSerrano) [reply src/modules/Transactions/utils/Transaction.js:56]: done

File: src/modules/Transactions/utils/Transaction.js (hunk 7 above)
Comment (ClementeSerrano) [reply src/modules/Transactions/utils/Transaction.js:28]: its being updated

File: src/modules/SendToken/hooks/useInitializationFeeCalculator.js (hunk 5 above)
Comment (ClementeSerrano) [reply src/modules/SendToken/hooks/useInitializationFeeCalculator.js]: done

**Review by soroushm — APPROVED** [review]

**Pull request merged**